### DISCUSSION TITLE
It's posible to implement passport-saml library in NextJS?

### DISCUSSION THREAD

**Discussion by RicardoNRevs** [bergie/passport-saml#885] [Q&A]
Anyone have implemented this library with Nextjs instead express?
Currently I have a expressJS server who is responsible to redirect to the SSO page.

I have something like this and works perfect in expressJS as backend and react for frontend:

```javascript
var samlStrategy = new saml.Strategy(
  {
    callbackUrl: process.env.CALLBACK_URL,
    entryPoint: process.env.ENTRY_POINT,
    issuer: process.env.ISSUER,
    cert: process.env.CERTIFICATE,
  },
  function (profile, done) {
    return done(null, profile);
  },
);
app.use(passport.initialize()); // init passport on every route call.
app.use(passport.session()); // allow passport to use "express-session".

// passport lifecycle
passport.use(samlStrategy);
passport.serializeUser(function (user, done) {
  done(null, user);
});
passport.deserializeUser(function (user, done) {
  done(null, user);
});
```

There's any recommendation about how can I do the same in next? or I'm just misunderstanding the functionality of both tools?

Thanks a lot!

**Comment by srd90** [comment]
Full disclosure: I have no idea what Next.JS is, but out of curiosity:

IMHO proper place to start digging existing solution for your use case would be to look for solutions which combine Next.JS and Passport.JS i.e. question is: https://www.google.com/search?q=next.js+passport

From the search results few promising links:
1. https://github.com/auth0-blog/nextjs-passport
2. https://auth0.com/blog/ultimate-guide-nextjs-authentication-auth0/
3. https://stackoverflow.com/a/63551398
4. https://github.com/vercel/next.js/tree/v14.1.1/examples/with-passport
5. https://github.com/vercel/next.js/tree/v14.1.1/examples/with-passport-and-next-connect

and while reading e.g. above pages think about passport-saml passport strategy to places where they use some other passport strategies. Auth0 blog's solution seems to use some legacy mode of next.js to use vanilla passport strategy while two other links - along with examples from next.js repository - seems to use some `next-connect` library.

edit: added one more link
